### PR TITLE
[CBR-254] Implement wallet ID in error

### PR DESCRIPTION
**Pull request by parsonsmatt**
## Description

This PR adds the wallet ID to the `WalletAlreadyExists` error type.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CBR-254#tab=Comments

## Type of change

- [] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [] 🔨 New or improved tests for existing code
- [] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: wallet/src/Pos/Wallet/Web/Methods/Restore.hs [hunk 1]
```diff
@@ -55,47 +58,95 @@ import           UnliftIO (MonadUnliftIO)
...
+
+-- | Creates a wallet. If the wallet with the given passphrase already exists,
+-- then we return 'Left' of the wallet's ID.
 mkWallet
     :: L.MonadWalletLogic ctx m
-    => PassPhrase -> CWalletInit -> Bool -> m (EncryptedSecretKey, CId Wal)
+    => PassPhrase
+    -> CWalletInit
+    -> Bool
+    -> m (Either (CId Wal) (EncryptedSecretKey, CId Wal))
```
Comment (KtorZ) [inline wallet/src/Pos/Wallet/Web/Methods/Restore.hs]: I'll be tempted to wrap that `Left` side in a newtype to make the semantic clear at the type-level. Don't you think?

File: wallet/src/Pos/Wallet/Web/Methods/Restore.hs [hunk 2]
```diff
@@ -55,47 +58,95 @@ import           UnliftIO (MonadUnliftIO)
...
 
     skey <- genSaveRootKey passphrase cwBackupPhrase
     let cAddr = encToCId skey
 
-    CWallet{..} <- L.createWalletSafe cAddr cwInitMeta isReady
-    -- can't return this result, since balances can change
+    eresult <- fmap Right (L.createWalletSafe cAddr cwInitMeta isReady)
+        `catch` \(e :: WalletError) ->
+            case e of
+                RequestError msg | msg == mnemonicExists ->
```
Comment (KtorZ) [inline wallet/src/Pos/Wallet/Web/Methods/Restore.hs]: How hard would it be to make `createWalletSafe` actually safe :joy:  ?

File: wallet/src/Pos/Wallet/Web/Methods/Restore.hs (hunk 1 above)
Comment (parsonsmatt) [reply wallet/src/Pos/Wallet/Web/Methods/Restore.hs]: hmmm, I don't know how much error-type-modification we want to do to the old wallet code. I could add a `newtype WalletAlreadyExists = WalletAlreadyExists (CId Wal)` but that'd overlap with what's in the new wallet. Or I could just use the old error type in the new code. Thoughts?

File: wallet/src/Pos/Wallet/Web/Methods/Restore.hs (hunk 2 above)
Comment (parsonsmatt) [reply wallet/src/Pos/Wallet/Web/Methods/Restore.hs]: Very easy! But it'd require touching a lot of the old wallet code that calls this function. This seemed like the easiest way to implement this in the least invasive way possible.

**Pull request merged**